assets/js/modules/analytics-4/components/dashboard/ActivationBanner/index.js

```diff
@@ -61,7 +70,43 @@ export default function ActivationBanner() {
 		)
 	);
 
-	const { setValues } = useDispatch( CORE_FORMS );
+	const accountID = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS ).getAccountID()
+	);
+
+	// Collection of selectors which the Setup Banner depends on.
+	const setupBannerSelectors = useMemo(
+		() => ( {
+			getProperties: { store: MODULES_ANALYTICS_4, args: [ accountID ] },
+			getAccounts: { store: MODULES_ANALYTICS },
+			getSettings: { store: MODULES_ANALYTICS_4 },
+		} ),
+		[ accountID ]
+	);
+
+	const setupBannerErrors = useSelect( ( select ) => {
+		const errors = {};
+
+		Object.keys( setupBannerSelectors ).forEach( ( selector ) => {
+			const { store, args } = setupBannerSelectors[ selector ];
+
+			const error = select( store ).getErrorForSelector(
+				selector,
+				args || []
+			);
+
+			if ( error ) {
+				errors[ selector ] = error;
+			}
+		} );
+
+		return errors;
```

Comment:
I think it's cleaner to use `Array.reduce()` here. However, it's just my personal preference. You can take it or ignore it if you think the current API is better. WDYT?

```suggestion
		return Object.keys( setupBannerSelectors ).reduce(
			( acc, selector ) => {
				const { store, args } = setupBannerSelectors[ selector ];

				const error = select( store ).getErrorForSelector(
					selector,
					args || []
				);

				return {
					...acc,
					...( error ? { [ selector ]: error } : {} ),
				};
			},
			{}
		);
```

Reply:
Reduce looks much cleaner. Updated, thank you!